backend/infrastructures/apps/load/views/views.py

```diff
@@ -131,25 +218,89 @@ def list(
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+        filters = query_ports.DataFilters(
+            is_satisfied=is_satisfied,
+            timestamp_from=request.query_params.get("timestamp_from"),
+            timestamp_to=request.query_params.get("timestamp_to"),
+        )
+        logger.info("Filters: %s", filters)
+
+        _ordering = ordering_dtos.Ordering.create_ordering(
+            request.query_params[common_consts.ORDERING_QUERY_PARAMETER_NAME].split(",")
+            if common_consts.ORDERING_QUERY_PARAMETER_NAME in request.query_params
+            else {}
         )
-        logger.info("Listed datasets.")
+        ordering = query_ports.DataOrdering(
+            full_name=_ordering.get("full_name"),
+            timestamp=_ordering.get("timestamp"),
+        )
+        logger.info("Ordering: %s", ordering)
+
+        try:
+            pagination = pagination_dtos.Pagination(
+                offset=request.query_params.get(
+                    common_consts.PAGINATION_OFFSET_QUERY_PARAMETER_NAME,
+                    pagination_dtos.PAGINATION_DEFAULT_OFFSET,
+                ),
+                records_per_page=request.query_params.get(
+                    common_consts.PAGINATION_LIMIT_QUERY_PARAMETER_NAME,
+                    pagination_dtos.PAGINATION_DEFAULT_LIMIT,
+                ),
+            )
+        except ValueError:
+            logger.warning("Invalid pagination parameters.")
+            return Response(
+                {common_consts.ERROR_DETAIL_KEY: "Invalid pagination parameters."},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+        logger.info("Pagination: %s", pagination)
+
+        try:
+            results: list[queries.OutputData]
+            count: int
+            results, count = queries.list_data(
+                repository=query_data_repository,
+                filters=filters,
+                ordering=ordering,
+                pagination=pagination,
+            )
+        except django_exceptions.ValidationError:
+            return Response(
+                {common_consts.ERROR_DETAIL_KEY: "Invalid query parameters."},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+        logger.info("Dataset listed successfully.")
 
         return Response(
-            data={
-                "count": count,
-                "data": OutputDataReadSerializer(output_data, many=True).data,
-            },
+            data=common_pagination.make_paginated_response(
+                url=request.build_absolute_uri(),
+                count=count,
+                offset=pagination.offset,
+                records_per_page=pagination.records_per_page,
+                results=[
+                    OutputDataReadSerializer(output_data).data
+                    for output_data in results
+                ],
+            ).data,
             status=status.HTTP_200_OK,
         )
+
+
+def _str_to_bool(value: str) -> bool | None:
```

Comment:
infrastructures/apps/common/parsers.py

```python
def map_bool_query_parameter_to_bool(value: str | None) -> bool | None:
    return value.lower() in ["true", "1", "yes"] if value is not None else None
```

Reply:
`_str_to_bool` to be removed.

Reply:
I did what you asked for here. but I changed the implementation to meet my requirement (i.e. when query parameter doesn't met true or false , it raises ValueError

please have a look in the next PR check iteration, thanks